packages/cli/src/commands/setup/ui/tasks/setup-component-library.js

```diff
@@ -0,0 +1,211 @@
+import fs, { promises as asyncfs } from 'fs'
+import path from 'path'
+
+import { getPaths } from '../../../../lib'
+
+/**
+ * Asynchronously creates a file at the specified path with the provided content, join()ed with '\n'
+ * If overwrite is false, and the file already exists, throws `Error(alreadyExistsError)`
+ * @param {string} path File path at which to create the file
+ * @param {Array} contentLines Array of lines to join and write into the file.
+ * @param {Boolean} overwrite Indicates if the file should be overwritten, if it already exists.
+ * @param {string} alreadyExistsError Message to throw if !overwrite && file already exists.
+ * // TODO: this seems like too general of a function to belong here. Where should it go?
```

Comment:
Have you found other places in the RW codebase where we use a pattern similar to what you have here? Maybe that could guide us in figuring out where to place this function.

Reply:
Not sure if this was your intention, but in the process of trying to answer this question I found `writeFile` in the `lib` package. I've replaced the uses here and deleted `createFile`.